terraform/README.md

```diff
@@ -74,3 +74,15 @@ When instantiating a new GKE cluster, you will want to merge its configuration i
 ## Formatting
 
 Our continuous integration is set up to do basic validation of Terraform files on pull requests. Besides any other testing, make sure to run `terraform fmt --recursive` or you will get build failures!
+
+## Autoscaling
+
+We use [Kubernetes Horizontal Pod Autoscalers](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) (HPAs) to automatically control the size of worker pools that process `intake-batch` and `aggregate` tasks. HPAs can manage deployment sizes based on a variety of different metric types. In our case, the metric of interest is the number of undelivered messages in the GCP PubSub subscriptions for the intake and aggregate tasks.
+
+GCP Stackdriver exposes metrics for PubSub, but to make those metrics visible to Kubernetes components, we deploy Google's [Custom Metrics Stackdriver Adapter](https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/custom-metrics-stackdriver-adapter), which is an implementation of the [external.metrics.k8s.io API](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#support-for-metrics-apis) that exposes Stackdriver metrics (see the `custom_metrics_adapter` module). Now that Kubernetes can see GCP metrics, we can configure HPAs to watch those metrics.
+
+We create an HPA for each Kubernetes Deployment (see the `kubernetes` Terraform module for details). We set `min_replicas` to 3 and `max_replicas` to 50. A minimum of three ensures we have reasonable minimum capacity at all times, while a maximum of 50 ensures that that we can scale up in response to really extraordinary spikes of workload while limiting the risk of flooding the Kubernetes cluster with more pods than it can possibly schedule.
```

Comment:
Isn't max_replicas of 50 high? If I recall correctly, we've never needed more than 10, and it seems like if we're going above 10 there might be a problem that we should look at.

Reply:
We have more than 10 intake batch workers in several states already, and 25 in California. 50 was chosen to be higher than we'd ever be likely to need but low enough that in the case of a runaway autoscale we don't take down the cluster. For the time being I'm optimizing for simplicity, hence the one-size-fits-all solution with one set of parameters for all localities. If we find this to be inefficient or unreliable then we can figure out more complex solutions, but as ever I intend to do the stupidest thing I can get away with.

I do like your idea of at least notifying us if we scale past some threshold. #508

Reply:
Ah I see! Thanks!